File: DSA/Math/0007_Reverse_Integer/solution.cpp
// 0007. Reverse Integer
// Difficulty : Medium
// Tags       : math
// Solved on  : 2018-11-28
// LeetCode   : https://leetcode.com/problems/reverse-integer/

class Solution {
public:
    int reverse( int x) {
      
       
       int n=0;
        
        while(x){
            int r=x%10;
            x=x/10;
            if(n>INT_MAX/10 || (n==INT_MAX/10 && r>7))
                return 0;
            if(n<INT_MIN/10 || (n==INT_MIN/10 && r<-8))
                return 0;
            n=n*10+r;
            
        }
        return n;
        
    }
};

File: DSA/Math/0007_Reverse_Integer/solution.js
// 0007. Reverse Integer
// Difficulty : Medium
// Tags       : math
// Solved on  : 2026-04-20
// LeetCode   : https://leetcode.com/problems/reverse-integer/

/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var sign = x < 0 ? -1 : 1;
    var num = Number(Math.abs(x).toString().split("").reverse().join(""));
    const INT_MAX = (2 ** 31) - 1;
    return num > INT_MAX ? 0 : sign * num;

};